packages/manifold/contracts/edition/ManifoldERC721Edition.sol

```diff
@@ -29,101 +29,108 @@ contract ManifoldERC721Edition is CreatorExtension, ICreatorExtensionTokenURI, I
-    function maxSupply(address creator, uint256 series) external view override returns(uint256) {
-        return _maxSupply[creator][series];
-    }
-
-    /**
-     * @dev See {IManifoldERC721Edition-createSeries}.
-     */
-    function createSeries(address creator, uint256 maxSupply_, string calldata prefix) external override creatorAdminRequired(creator) returns(uint256) {
-        _currentSeries[creator] += 1;
-        uint256 series = _currentSeries[creator];
-        _maxSupply[creator][series] = maxSupply_;
-        _tokenPrefix[creator][series] = prefix;
-        emit SeriesCreated(msg.sender, creator, series, maxSupply_);
-        return series;
-    }
-
-    /**
-     * @dev See {IManifoldERC721Edition-latestSeries}.
-     */
-    function latestSeries(address creator) external view override returns(uint256) {
-        return _currentSeries[creator];
+    function maxSupply(address creatorCore, uint256 instanceId) external view override returns(uint256) {
+        if (_maxSupply[creatorCore][instanceId] == 0) revert("Invalid instanceId");
+        return _maxSupply[creatorCore][instanceId];
     }
 
     /**
      * See {IManifoldERC721Edition-setTokenURIPrefix}.
      */
-    function setTokenURIPrefix(address creator, uint256 series, string calldata prefix) external override creatorAdminRequired(creator) {
-        require(series > 0 && series <= _currentSeries[creator], "Invalid series");
-        _tokenPrefix[creator][series] = prefix;
+    function setTokenURIPrefix(address creatorCore, uint256 instanceId, string calldata prefix) external override creatorAdminRequired(creatorCore) {
+        if (_maxSupply[creatorCore][instanceId] == 0) revert("Invalid instanceId");
+        _tokenPrefix[creatorCore][instanceId] = prefix;
     }
     
     /**
      * @dev See {ICreatorExtensionTokenURI-tokenURI}.
      */
-    function tokenURI(address creator, uint256 tokenId) external view override returns (string memory) {
-        (uint256 series, uint256 index) = _tokenSeriesAndIndex(creator, tokenId);
-        return string(abi.encodePacked(_tokenPrefix[creator][series], (index+1).toString()));
+    function tokenURI(address creatorCore, uint256 tokenId) external view override returns (string memory) {
+        (uint256 instanceId, uint256 index) = _tokenInstanceAndIndex(creatorCore, tokenId);
+        return string(abi.encodePacked(_tokenPrefix[creatorCore][instanceId], (index+1).toString()));
     }
     
     /**
      * @dev See {IManifoldERC721Edition-mint}.
      */
-    function mint(address creator, uint256 series, address recipient, uint16 count) external override nonReentrant creatorAdminRequired(creator) {
-        require(count > 0, "Invalid amount requested");
-        require(_totalSupply[creator][series]+count <= _maxSupply[creator][series], "Too many requested");
+    function mint(address creatorCore, uint256 instanceId, address recipient, uint16 count) external override nonReentrant creatorAdminRequired(creatorCore) {
```

Comment:
Why do we need `nonReentrant` here ? Curious cause we don't have this in our other contract mint function such as `LazyClaims721` contract

Reply:
I believe it's because of the order of the end of the function here...

```
        uint256[] memory tokenIds = IERC721CreatorCore(creatorCore).mintExtensionBatch(recipient, count);
        _updateIndexRanges(creatorCore, instanceId, tokenIds[0], count);
```

If the recipient is a smart contract address, and has `onERC721Received` function that calls back in here, it could continue to loop this function before we call `_updateIndexRanges` to account for the total minted..

@wwhchung might know more